app/src/main/java/com/example/gradeassist/ui/home/HomeFragment.kt

```diff
@@ -28,4 +35,22 @@ class HomeFragment : Fragment() {
         })
         return root
     }
+
+    private fun logon() {
+        var providers = arrayListOf(
+            AuthUI.IdpConfig.EmailBuilder().build()
+        )
+        startActivityForResult(
+            AuthUI.getInstance().createSignInIntentBuilder().setAvailableProviders(providers).build() , AUTH_REQUEST_CODE
+        )
+    }
```

Comment:
Wrote a logon Function in the main home fragment, users need to sign in to get their quizzes and results.